solr/core/src/java/org/apache/solr/core/RateLimiterConfig.java

```diff
@@ -31,6 +33,7 @@ public class RateLimiterConfig {
   public int allowedRequests;
   public boolean isSlotBorrowingEnabled;
   public int guaranteedSlotsThreshold;
+  public List<RateLimiterPayload.ReadBucketConfig> readBuckets;
```

Comment:
My guess of `readBuckets` nested inside RateLimiterConfig is to allow the json reader to handle both cases:
1. no bucket, just configuration on the existing non bucket limiter:
```
"rate-limiters": {
     "allowedRequests": 2,
     "slotAcquisitionTimeoutInMS": 1000
}
```
2. use bucket:
"rate-limiters": {
  "readBuckets": [
    {
      "name": "test-bucket",
        "conditions": [{
          "queryParamPattern": {
            "q-bucket": "test-bucket"
          }
        }],
        "allowedRequests": 2,
        "slotAcquisitionTimeoutInMS": 1000
     }]
}

However, this could introduce some confusion as properties such as `allowedRequests` could be defined either under each "readBuckets" and/or "rate-limiters".

The `allowedRequests` defined under "rate-limiters" will be ignored if there are buckets. It seems all fields in `RateLimiterConfig` will be ignored except `readBuckets`, which lead me to think whether we should just use 2 separate class/config to avoid confusion and unused fields.

Could we just use something like check existence of `rate-limiters` and `rate-limiters-buckets` key to identify whether it's bucket and non-bucket use case (I think they are mutually exclusive) For example:
1. No bucket usage
```
"rate-limiters": {
     "allowedRequests": 2,
     "slotAcquisitionTimeoutInMS": 1000
}
```
or
2. Bucket usage
```
"rate-limiters-buckets" : [
  {
    "name": "test-bucket",
    "conditions": [{
      "queryParamPattern": {
        "q-bucket": "test-bucket"
       }
     }],
     "allowedRequests": 2,
     "slotAcquisitionTimeoutInMS": 1000
   }
]
```

With this proposal the first case will be deserialized as `RateLimiterConfig`, and the 2nd case will just be parsed as an array of `ReadBucketConfig`, which might affect the ctor signature of `RequestRateLimiter` (and child classes), more on this later.

Reply:
I thought about it. This was done to simplify the code.